test/interchain/inter_chain_accounts_test.go

```diff
@@ -0,0 +1,119 @@
+			Chain1:  celestia,
+			Chain2:  cosmosHub,
+			Relayer: relayer,
+			Path:    pathName,
+		})
+
+	ctx := context.Background()
+	reporter := testreporter.NewNopReporter().RelayerExecReporter(t)
+	err := interchain.Build(ctx, reporter, interchaintest.InterchainBuildOptions{
+		TestName:  t.Name(),
+		Client:    client,
+		NetworkID: network,
+	})
+	require.NoError(t, err)
+	t.Cleanup(func() { _ = interchain.Close() })
+
+	err = relayer.StartRelayer(ctx, reporter, pathName)
+	require.NoError(t, err)
+
+	err = testutil.WaitForBlocks(ctx, 2, celestia, cosmosHub)
+	require.NoError(t, err)
+
+	celestiaConnections, err := relayer.GetConnections(ctx, reporter, celestia.Config().ChainID)
+	require.NoError(t, err)
+	require.Len(t, celestiaConnections, 1)
+
+	cosmosConnections, err := relayer.GetConnections(ctx, reporter, cosmosHub.Config().ChainID)
+	require.NoError(t, err)
+	require.Len(t, cosmosConnections, 2) // 2 connections: the first is connection-0 and the second is connection-localhost.
+	cosmosConnection := cosmosConnections[0]
+
+	users := interchaintest.GetAndFundTestUsers(t, ctx, t.Name(), math.NewInt(10_000_000_000), celestia, cosmosHub)
+	err = testutil.WaitForBlocks(ctx, 2, celestia, cosmosHub)
+	require.NoError(t, err)
+
+	celestiaUser, cosmosUser := users[0], users[1]
+	celestiaAddr := celestiaUser.(*cosmos.CosmosWallet).FormattedAddressWithPrefix(celestia.Config().Bech32Prefix)
+	cosmosAddr := cosmosUser.(*cosmos.CosmosWallet).FormattedAddressWithPrefix(cosmosHub.Config().Bech32Prefix)
+	fmt.Printf("celestiaAddr: %s, cosmosAddr: %v\n", celestiaAddr, cosmosAddr)
+
+	registerICA := []string{
+		cosmosHub.Config().Bin, "tx", "interchain-accounts", "controller", "register", cosmosConnection.ID,
+		"--chain-id", cosmosHub.Config().ChainID,
+		"--home", cosmosHub.HomeDir(),
+		"--node", cosmosHub.GetRPCAddress(),
+		"--from", cosmosUser.KeyName(),
+		"--keyring-backend", keyring.BackendTest,
+		"--fees", fmt.Sprintf("300000%v", cosmosHub.Config().Denom),
+		"--gas", "300000", // the auto gas estimation underestimates the gas required.
+		"--yes",
+	}
+	stdout, stderr, err := cosmosHub.Exec(ctx, registerICA, nil)
+	require.NoError(t, err)
+	require.Empty(t, stderr)
+	t.Logf("stdout %v", string(stdout))
+
+	err = testutil.WaitForBlocks(ctx, 5, celestia, cosmosHub)
+	require.NoError(t, err)
+
+	queryICA := []string{
+		cosmosHub.Config().Bin, "query", "interchain-accounts", "controller", "interchain-account", cosmosAddr, cosmosConnection.ID,
+		"--chain-id", cosmosHub.Config().ChainID,
+		"--home", cosmosHub.HomeDir(),
+		"--node", cosmosHub.GetRPCAddress(),
+	}
+	stdout, stderr, err = cosmosHub.Exec(ctx, queryICA, nil)
+	require.NoError(t, err)
+	require.Empty(t, stderr)
+	t.Logf("stdout %v\n", string(stdout))
+	assert.NotEmpty(t, string(stdout))
+
+	channels, err := relayer.GetChannels(ctx, reporter, cosmosHub.Config().ChainID)
+	require.NoError(t, err)
+	fmt.Printf("cosmosHub channels: %#v\n", channels)
```

Comment:
Do we still need the print statements?

Reply:
Oops not really needed anymore. I accidentally left them in from debugging.

Reply:
fb3ca28eb3b6068d7593d8cee7ec88219a637c77